lang/src/org/partiql/lang/eval/physical/operators/AggregateOperatorFactory.kt

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *      http://aws.amazon.com/apache2.0/
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ *  language governing permissions and limitations under the License.
+ */
+package org.partiql.lang.eval.physical.operators
+
+import com.amazon.ion.IonValue
+import org.partiql.lang.domains.PartiqlPhysical
+import org.partiql.lang.eval.DEFAULT_COMPARATOR
+import org.partiql.lang.eval.ExprValueType
+import org.partiql.lang.eval.MISSING_ANNOTATION
+import org.partiql.lang.eval.physical.EvaluatorState
+import org.partiql.lang.eval.physical.SetVariableFunc
+import org.partiql.lang.eval.relation.relation
+import org.partiql.lang.util.toIntExact
+import java.util.TreeSet
+
+public abstract class AggregateOperatorFactory(name: String) : RelationalOperatorFactory {
+    public final override val key: RelationalOperatorFactoryKey = RelationalOperatorFactoryKey(RelationalOperatorKind.AGGREGATE, name)
+    public abstract fun create(
+        impl: PartiqlPhysical.Impl,
+        source: RelationExpression,
+        keys: List<CompiledGroupKey>,
+        functions: List<CompiledAggregateFunction>
+    ): RelationExpression
+
+    public abstract fun getFunctionFactory(
+        operator: AggregationKind,
+        name: String
+    ): AggregationFunctionFactory
+}
+
+internal class FullAggregateFactory(name: String) : AggregateOperatorFactory(name) {
```

Comment:
Seems like a hash based approach? Mind explain a little more on the meaning of `FullAggregate`?

Reply:
Well, instead of performing a partial aggregation and then a final aggregation, I opted to create a "Full Aggregation" as it's easier to implement. Just some naming thing, but I'm not pressed on keeping the name.